Porting visualizations to new gallery

suhaani-agarwal opened the pull request:
porting the following visualizations from https://rcdata.nau.edu/genomic-ml/animint-gallery/ to https://animint.github.io/gallery/

55 - [American Temperature Maps](https://suhaani-agarwal.github.io/central-american-temperature-maps/) , old viz : [link](https://rcdata.nau.edu/genomic-ml/animint-gallery/2014-10-09-central-america-temperature-maps/index.html)
41 - [World Bank paper](https://suhaani-agarwal.github.io/WorldBank-paper/) , old viz : [link](https://rcdata.nau.edu/genomic-ml/animint-gallery/2016-03-11-WorldBank-data-viz-for-Animint-paper/index.html)
35 - [ChIP seq data unaligned viz](https://suhaani-agarwal.github.io/chip-seq-unaligned/), old viz : [link](https://rcdata.nau.edu/genomic-ml/animint-gallery/2016-08-18-chip-seq-unaligned/index.html)

here is the preview in my forked version of the gallery : https://suhaani-agarwal.github.io/gallery/

Ill make the demonstration videos once the visualization outlooks are finalized.

suhaani-agarwal committed:
update 2025-04-10 15:58:55.939713

suhaani-agarwal committed:
update 2025-05-10 13:57:17.383153

suhaani-agarwal committed:
update 2025-05-10 15:08:53.120849

tdhock commented:
Hi! thanks, looks like  good start.
https://suhaani-agarwal.github.io/WorldBank-paper/ seems to be missing aes(key) for geom_point, because I see a strange transition for some years, for example between 82 and 83. please fix.

tdhock commented:
on this one, the violet line for selected seems to be semi-transparent. can you please try to make it opaque for emphasis? (alpha=1, alpha_off=0.3 or something?)
![image](https://github.com/user-attachments/assets/3a06682f-defa-4577-bac3-7786d026c260)
also please add aes(key) for the geom_points

tdhock commented:
the grey background here makes it hard to see some data points. can you please change to theme_bw()? 
![image](https://github.com/user-attachments/assets/60ac6c55-68c3-40ed-8f5d-3f9a7fa89b1f)

suhaani-agarwal commented:
made all changes. please check.
55 - [American Temperature Maps](https://suhaani-agarwal.github.io/central-american-temperature-maps/) , 
41 - [World Bank paper](https://suhaani-agarwal.github.io/WorldBank-paper/) , 
35 - [ChIP seq data unaligned viz](https://suhaani-agarwal.github.io/chip-seq-unaligned/)

tdhock commented:
for this one the y axis title is unreadable (should be able to see Celsius, please adjust text content or size).
also for the scatter plots it is difficult to see which point is selected. Can you please make the selected violet as in the lines in the other plots?
Also there seems to be a lack of smooth transitions for the geom_tiles --can you please fix? do you need to add aes(key) ?
![image](https://github.com/user-attachments/assets/b8816d78-d4a0-4d70-9607-90d4523fe827)

tdhock commented:
the axis text is unreadable, please fix.
- (mega base ...
- (percent...
- annotations (p...
also the text in select samples is a bit too large, so they are overlapping. can you please reduce the text size, or make the plot size bigger, so the text between adjacent sample names does not overlap?
![image](https://github.com/user-attachments/assets/d215726b-91c9-4cd1-9ff1-28ffa88033f3)

suhaani-agarwal commented:
> the axis text is unreadable, please fix.
> for this one the y axis title is unreadable (should be able to see Celsius, please adjust text content or size).

I have made these changes, although in my system, the entire y axis was visible, I have made the plots a little bigger and changed some alignments to make the labels appear fully.

>   also the text in select samples is a bit too large, so they are overlapping. can you please reduce the text size, or make the plot size bigger, so the text between adjacent sample names does not overlap?

I have solved this error, made the plot size bigger.

> also for the scatter plots it is difficult to see which point is selected. Can you please make the selected violet as in the lines in the other plots?

This is resolved now.

> Also there seems to be a lack of smooth transitions for the geom_tiles --can you please fix? do you need to add aes(key) ?

I have already added aes(key) in geom_tiles , the smooth transitions for the colours are still not happening.
```surftempMap = ggplot() + 
    theme_bw() +
    theme_animint(width=420, height=450) +
    geom_tile(aes(x = long, y = lat, fill = surftemp, key = id)),
            clickSelects = "id", showSelected = c("time2"),
            data = climate) + 
    scale_fill_gradient2("deg. C", low = "blue", mid = "white", high = "#ff0000",
                       midpoint = 0, limits = lims$surftemp) + 
    ggtitle("Surface temperature") +
    geom_path(aes(long, lat, group = group), col = "#908f8f",
            data = countries) +

tdhock commented:
In your R code the following is un-necessary (does not do anything, can be confusing for the reader), please delete.
```r
    hover = list(
      size = 2,
      alpha = 1
    )
...
    # smooth = TRUE,
    smooth = list(
    method = "linear",
    duration = 2000
  ),
    interpolate = list(
    surftemp = "linear",
    surfdev = "linear")
  ),
```

tdhock commented:
Overall the viz looks better but we need to figure out why the tile smooth transitions are not happening (they should).

Generally I think it would be easier to understand for the reader, if there was one selection color for the region/tile (purple) and another selection color for the time point (maybe blue?).
Currently scatterplot shows emphasis using only purple, can you please change the time one to blue?
![image](https://github.com/user-attachments/assets/9b3b121e-3926-434e-9622-ec9e0498b44f)


Similarly the time points are selected with the tallrect here which is black, can you please change to blue?
![image](https://github.com/user-attachments/assets/01f8b579-771b-4028-9c5c-f80dd1fa10f1)

tdhock commented:
You show the highlighted region and time point by adding a line or point with additional showSelected. 
```r
  geom_line(
    aes_string(x = "time2", y = var.names[[1]], group = "id"),
    data = climate,
    colour = "black",
    size = 1.5,
    alpha = 0.55,
    clickSelects = "id",
    hover = list(
      size = 2,
      alpha = 1
    )
  ) +
  geom_line(
    aes_string(x = "time2", y = var.names[[1]], group = "id"),
    data = climate,
    colour = selected.color, # Purple for selected
    size = 3,             # Slightly bolder when selected
    alpha = 2,              # Opaque when selected
    showSelected = "id"     # Only show for selected id
  ),  # Actual selection,
```
This is fine, but please add a smooth transition for the selected lines/points.
The second geom_line would need `aes_string(key="column_with_constant_value")`.
If you don't want smooth transitions between the highlighted line/point then you would delete the second geom_line and instead use geom_line(color_off) in the first one.

Also
- it does not need aes(group) because there is only one displayed at at time, so please remove that, because un-necessary code can be confusing (make it harder for the reader to understand).
- alpha should be between 0 and 1 but you have 2 here, so please change to 1 to indicate full opacity.

suhaani-agarwal commented:
I have made the following changes : 
- changed the colours of the selected time components (blue) including the tallrect that changes with time in the time series plot and the geom_points in the scatterHere plot that shows selection for time.
- added smooth transitions for the selected lines and points.
- pushed the cleaner source code after removing unnecessary code.
- added some help parameters in the code for the tour feature.
----
- will figure out why the geom_tiles smooth transitions arent working
- shall I add a legend to the time series plots indicating blue colour for selected time and purple for region ?

tdhock commented:
this looks better.
![image](https://github.com/user-attachments/assets/1eea1a3f-e3a2-422b-8771-09b53bfb08bb)

> shall I add a legend to the time series plots indicating blue colour for selected time and purple for region ?

that is a great idea!! probably would require attaching a new column to the different geoms as below,

```{r}
geom_line(aes(
  color=select),
  data=data.table(select="region"))+
geom_tallrect(aes(
  color=select),
  data=data.table(select="month"))
```

also could you please make sure the geom_tile/geom_point/geom_line which are used to select region all have the same color. Currently geom_tile is black but others are violet. Not sure if either of these is a great idea .. try experimenting with some different colors to see what looks good across the whole data viz?

suhaani-agarwal commented:
I have made some changes to the selected colours (neon green now for the selected region everywhere) and also added a legend for indicating selected colour to the timeSeries plot according to the method you suggested. 
Is the legend okay? because right now it is clickable. it shouldn't be because it only indicates the colours. can I make it non interactive like the legend in the geom_tile maps ? 

Also, if the other plots are okay, shall I make the demonstration videos for them?

tdhock commented:
hi! the green looks good!

tdhock commented:
how did you get the legend to be clickable but not have any response in the plot when you click?
please add showSelected="Selected" in corresponding geoms so we see a response.

tdhock commented:
please remove one of the two new Selected legends (repetitive legends can be confusing, typically you should only keep the one on the right).

tdhock commented:
for time variable, you currently have time2=1996.415 etc (decimal values). Can you please convert to factors like 1996-04?

tdhock commented:
geom_tile still has no smooth transitions. Can you please try making a minimal reproducible example to try to figure out if this is a bug? (this example is pretty complex)
also can you please check test cases to see if smooth transitions in geom_tile are tested?

tdhock commented:
please fix these issues before making a video.

suhaani-agarwal commented:
> how did you get the legend to be clickable but not have any response in the plot when you click?
> please add showSelected="Selected" in corresponding geoms so we see a response.

> please remove one of the two new Selected legends (repetitive legends can be confusing, typically you should only keep the one on the right).

I have changed the position of the legend as earlier it was appearing to be a little confusing and clustered. It is working and is clickable now.

> for time variable, you currently have time2=1996.415 etc (decimal values). Can you please convert to factors like 1996-04?

For this, i have added tooltips of this format (yyyy-mm) for the time variable that appear on hovering over the area of tallrect. Is that okay?

the time2 being shown in selection menu cant be changed, because we need the numerical time2 value for the same look of the plots. I tried changing the time2 variable values by writing a function as follows : 
```convert_to_ym <- function(decimal_years) {
  sapply(decimal_years, function(x) {
    year <- floor(x)
    decimal_part <- x - year
    month <- round(decimal_part * 12) + 1  # Convert decimal to month (1-12)
    month <- pmin(pmax(month, 1), 12)  # Ensure month stays between 1-12
    year + (month / 100)  # Creates yyyy.mm format
  })
}

# Apply the conversion to your data
climate$time2 <- convert_to_ym(climate$time2)
```
```
# Example conversions:
# 1995.00 → 1995.01 (January)
# 1995.08333 → 1995.02 (February) 
# 1995.91666 → 1995.12 (December)
```

This formatted the time2 values for all show selected and also for the selected menu , but the viz then appeared like this
<img width="1421" alt="image" src="https://github.com/user-attachments/assets/ae39a73e-e4bb-4590-bb97-74988e909454" />

Is it okay if the time2 in show selected remains the same and if the user wants to know what time2 decimal is representing, they can access the tooltips, the labels on top of the temperature maps, or the text on the scatter plot? If it still needs to be changed, please let me know

tdhock commented:
Hi! that is better, thanks.
but your date format conversion code is too complicated. please try this instead:
```r
> x <- 1995.08333;sprintf("%d-%02d",round(x), round((x %% 1) * 12))
[1] "1995-01"
```

suhaani-agarwal commented:
**Smooth transitions for geom_tiles have now been implemented**  Please check - [American Temperature Maps](https://suhaani-agarwal.github.io/central-american-temperature-maps/) 

for this, I made two geom_tiles as follows : 

```
#BASE LAYER SMOOTH TRANSITIONS
    geom_tile(aes_string(x="long", y="lat", fill=var.name, key = "id"
                  ), showSelected="time2", color = NA, alpha = 1,
              data=climate)+ 
#SELECTION LAYER     
    geom_tile(aes_string(x="long", y="lat", key = "id"
                  ),clickSelects = "id", color = selected.color.region , fill=NA, alpha = 0.8, showselected="time2",
              data=climate)+
```

Because the selection of region (clickSelects) and smooth transitions cant happen simultaneously. So I made two layers, one for the smooth transitions having show selected = "time2" that doesnt have a border colour but shows the filled colours . and the other layer having clickselects = "id" doesnt have filled colours but border colour is of the selected region.

tdhock commented:
This looks great thanks! please add a video.

suhaani-agarwal committed:
update 2025-05-24 00:01:10.605292

suhaani-agarwal committed:
update 2025-05-24 00:08:17.179265

tdhock commented:
thanks for the video link but it does not work for me. I get the page below which asks for login. Can you please fix so that login is not required?
![image](https://github.com/user-attachments/assets/85e4d87e-d7cb-4089-9a40-b44100cfd7ae)

suhaani-agarwal commented:
I have updated the content rating on vimeo , please confirm if you are being able to open the video now without logging in

tdhock commented:
great! I watched the video and I think you did a fantastic job! (probably even better than I could have)
Thanks very much!!

suhaani-agarwal commented:
Thank you! Glad it worked well, appreciate your guidance!
